clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java

```diff
@@ -369,24 +393,40 @@ private void maybeTriggerWakeup() {
     }
 
     public void disableWakeups() {
-        wakeupDisabledCount++;
+        lock.lock();
```

Comment:
Yeah, if we don't use any of `ReentrantLock`'s additional functionality, then I'd suggest using `synchronized`. It's also worth noting that the latter is a better option for rarely contended locks. The VM has a bunch of optimisations for such cases. `ReentrantLock`, on the other hand, is optimised for cases with some contention (it tends to be competitive with `synchronized` for such cases).